modules/node/src/channel/channel.provider.ts

```diff
@@ -71,52 +68,43 @@ class ChannelMessaging extends AbstractMessagingProvider {
 
   async requestCollateral(pubId: string, data: { assetId?: string }): Promise<CFCoreTypes.DepositResult> {
     // do not allow clients to specify an amount to collateralize with
-    return this.channelService.requestCollateral(pubId, getAddress(data.assetId));
+    return (this.channelService.rebalance(
+      pubId,
+      getAddress(data.assetId),
+      RebalanceType.COLLATERALIZE,
+    ) as unknown) as CFCoreTypes.DepositResult;
   }
 
   async withdraw(pubId: string, data: { tx: CFCoreTypes.MinimalTransaction }): Promise<TransactionResponse> {
     return this.channelService.withdrawForClient(pubId, data.tx);
   }
 
-  async addPaymentProfile(
-    pubId: string,
-    data: {
-      assetId: string;
-      minimumMaintainedCollateral: string;
-      amountToCollateralize: string;
-    },
-  ): Promise<PaymentProfileRes> {
-    const {
-      amountToCollateralize,
-      minimumMaintainedCollateral,
-      assetId,
-    } = await this.channelService.addPaymentProfileToChannel(
+  async addRebalanceProfile(pubId: string, data: { profile: RebalanceProfile }): Promise<void> {
+    await this.channelService.addRebalanceProfileToChannel(
       pubId,
-      data.assetId,
-      bigNumberify(data.minimumMaintainedCollateral),
-      bigNumberify(data.amountToCollateralize),
+      data.profile.assetId,
+      bigNumberify(data.profile.lowerBoundCollateralize),
+      bigNumberify(data.profile.upperBoundCollateralize),
+      bigNumberify(data.profile.lowerBoundReclaim),
+      bigNumberify(data.profile.upperBoundReclaim),
     );
-
-    return convert.PaymentProfile("str", {
-      amountToCollateralize,
-      assetId,
-      minimumMaintainedCollateral,
-    });
   }
 
-  async getPaymentProfile(pubId: string, data: { assetId?: string }): Promise<PaymentProfileRes | undefined> {
-    const prof = await this.channelRepository.getPaymentProfileForChannelAndToken(pubId, data.assetId);
+  async getRebalanceProfile(pubId: string, data: { assetId?: string }): Promise<RebalanceProfile> {
+    const prof = await this.channelRepository.getRebalanceProfileForChannelAndAsset(pubId, data.assetId);
 
     if (!prof) {
       return undefined;
     }
 
-    const { amountToCollateralize, minimumMaintainedCollateral, assetId } = prof;
-    return convert.PaymentProfile("str", {
-      amountToCollateralize,
+    const { upperBoundReclaim, lowerBoundReclaim, upperBoundCollateralize, lowerBoundCollateralize, assetId } = prof;
+    return {
       assetId,
-      minimumMaintainedCollateral,
-    });
+      lowerBoundCollateralize: lowerBoundCollateralize.toString(),
+      lowerBoundReclaim: lowerBoundReclaim.toString(),
+      upperBoundCollateralize: upperBoundCollateralize.toString(),
+      upperBoundReclaim: upperBoundReclaim.toString(),
+    };
```

Comment:
could just use the conversion function

Reply:
We've talked about deprecating the conversion functions due to complexity, so I didn't do it here, but you're right, may as well use it.